api/src/main/java/com/epam/esm/controller/DefaultExceptionHandler.java

```diff
@@ -0,0 +1,108 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dao.exception.DaoException;
+import com.epam.esm.dao.exception.GiftNotFoundException;
+import com.epam.esm.dao.exception.TagNameRegisteredException;
+import com.epam.esm.dao.exception.TagNotFoundException;
+import com.epam.esm.model.ErrorResponse;
+import com.epam.esm.model.dto.GiftCertificateDto;
+import com.epam.esm.service.exception.ServiceException;
+import com.epam.esm.util.Status;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+@Slf4j
+public class DefaultExceptionHandler {
+    private final ObjectMapper objectMapper;
+    private final MessageSource messageSource;
+
+    private static final String SEARCH_ENTITY_ID_REGEX = "^.+/";
+
+    @Autowired
+    public DefaultExceptionHandler(ObjectMapper objectMapper, MessageSource messageSource) {
+        this.objectMapper = objectMapper;
+        this.messageSource = messageSource;
+    }
+
+    @ExceptionHandler(value = {
+            ServiceException.class,
+            DaoException.class
+    })
+    public ResponseEntity<ErrorResponse> handleConflict(RuntimeException ex, WebRequest webRequest, HttpServletRequest request ) {
+        String message = messageSource.getMessage(Status.DEFAULT.getCode().toString(), null, webRequest.getLocale());
+
+        return getErrorResponseResponseEntity(request, ex, webRequest, Status.DEFAULT, HttpStatus.INTERNAL_SERVER_ERROR, message);
+    }
+
+    @ExceptionHandler(GiftNotFoundException.class)
+    @SneakyThrows
+    public ResponseEntity<ErrorResponse> handleGiftNotFound(HttpServletRequest request, HttpServletResponse response, Exception ex, WebRequest webRequest) {
+        String message = messageSource.getMessage(Status.GIFT_NOT_FOUND.getCode().toString(), null, webRequest.getLocale());
+
+        String resultMessage = errorMessageNotFoundById(request, message);
```

Comment:
Why don't  you add field `giftId` into GiftNotFoundException? in such case you don't need to parse error message

Reply:
the same comment for TagNotFoundException